### PR TITLE
104033 Seed Id at 4000001. ItemNo=Id

### PR DESCRIPTION
**Pull request by janid1967**
[AB#104033](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/104033)

Punch ItemNo is removed as DB column. ItemNo is now a getter-prop and just return Id. 
Seed Id at 4000001
Punch Description is now a required field, given from client when create instead of ItemNo

(Migration created from start again)

### CODE REVIEW

File: src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs [hunk 1]
```diff
@@ -27,21 +27,22 @@ public async Task TestInitialize()
...
             TestFactory.ProjectGuidWithAccess);
 
         // Assert
         AssertValidGuidAndRowVersion(guidAndRowVersion);
         var newPunchItem = await PunchItemsControllerTestsHelper
             .GetPunchItemAsync(UserType.Writer, TestFactory.PlantWithAccess, guidAndRowVersion.Guid);
         Assert.IsNotNull(newPunchItem);
-        Assert.AreEqual(itemNo, newPunchItem.ItemNo);
+        Assert.IsTrue(!newPunchItem.Description.IsEmpty());
+        Assert.IsTrue(newPunchItem.ItemNo > 4000000);
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs:45]: Maybe add a comment or move to constant to give context to the magic number.

File: src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs [hunk 2]
```diff
@@ -70,7 +71,7 @@ public async Task GetAllPunchItemsInProject_AsReader_ShouldGetAllPunchItems()
 
         // Assert
         Assert.IsTrue(punchItems.Count > 0);
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs:73]: Comment: Have we verified it get all records if we get more than zero items in punchItems?

File: src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs [hunk 3]
```diff
@@ -70,7 +71,7 @@ public async Task GetAllPunchItemsInProject_AsReader_ShouldGetAllPunchItems()
...
-        Assert.IsTrue(punchItems.All(p => !p.ItemNo.IsEmpty()));
+        Assert.IsTrue(punchItems.All(p => !p.Description.IsEmpty()));
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs:74]: Given name of test: GetAllPunchItemsInProject_AsReader_ShouldGetAllPunchItems
This seems to be testing a different concern, namely whether the properties are empty or not.
Would it be more natural to move these two to a different test with a relevant name?

**Review by jsoi-eq — COMMENTED** [review]
Finished review

File: src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs (hunk 2 above)
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs:74]: renamed test and added comment

File: src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs (hunk 3 above)
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs]: removed check of properties, this should be done in unit tests anyway

**Pull request merged**